python/semantic_kernel/connectors/memory/astradb/astra_client.py

```diff
@@ -0,0 +1,160 @@
+        embedding_dim: int,
+        similarity_function: str,
+        session: Optional[aiohttp.ClientSession] = None
+    ):
+        self.astra_id = astra_id
+        self.astra_application_token = astra_application_token
+        self.astra_region = astra_region
+        self.keyspace_name = keyspace_name
+        self.embedding_dim = embedding_dim
+        self.similarity_function = similarity_function
+
+        self.request_base_url = f"https://{self.astra_id}-{self.astra_region}.apps.astra.datastax.com/api/json/v1/{self.keyspace_name}"
+        self.request_header = {
+            "x-cassandra-token": self.astra_application_token,
+            "Content-Type": "application/json",
+        }
+        self._session = session
+
+    async def _run_query(self, request_url: str, query: Dict):
+        async with AsyncSession(self._session) as session:
+            async with session.post(request_url, data=json.dumps(query), headers = self.request_header ) as response:
+                if response.status == 200:
+                    response_dict = await response.json()
+                    if "errors" in response_dict:
+                        raise Exception(f"Astra DB request error - {response_dict['errors']}")
+                    else:
+                        return response_dict
+                else:
+                    raise Exception(f"Astra DB not available. Status : {response}")
+
+    async def find_collections(self, include_detail: bool = True):
+        query = {"findCollections": {"options": {"explain": include_detail}}}
+        result = await (self._run_query(self.request_base_url, query))
+        return result["status"]["collections"]
+
+    async def find_collection(self, collection_name: str):
+        collections = await self.find_collections(False)
+        found = False
+        for collection in collections:
+            if collection == collection_name:
+                found = True
+                break
+        return found
+
+    async def create_collection(self, collection_name: str, embedding_dim: Optional[int] = None, similarity_function: Optional[str] = None):
+        query = {"createCollection": {
+            "name": collection_name,
+            "options": {
+                "vector": {
+                    "dimension": embedding_dim if embedding_dim is not None else self.embedding_dim,
+                    "metric": similarity_function if similarity_function is not None else self.similarity_function,
+                }
+            }
+        }}
+        result = await (self._run_query(self.request_base_url, query))
+        return True if result["status"]["ok"] == 1 else False
+
+    async def delete_collection(self, collection_name: str):
+        query = {"deleteCollection": {"name": collection_name}}
+        result = await (self._run_query(self.request_base_url, query))
+        return True if result["status"]["ok"] == 1 else False
+
+    def _build_collection_query(self, collection_name: str):
```

Comment:
this returns the collection_url not the query, please rename!

Reply:
Done